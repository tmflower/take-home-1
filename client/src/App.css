body {
   background-image: url("../public/background.jpg");
   background-size: cover;
   color: white;
   font-family: 'Varela Round', sans-serif;
}

a:link {
    color: white;
    text-decoration: none;
}

a:visited {
    color: white;
    text-decoration: none;
}

a:hover {
    color: white;
    text-decoration: none;
    scale: 1.5;
}

a:active {
    color: white;
    text-decoration: none;
}
