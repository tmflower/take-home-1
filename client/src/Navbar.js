import { NavLink } from "react-router-dom";
import { Home } from "./routes/Home";
import { Signup } from "./routes/Signup";
import { Admin } from "./routes/Admin";
import { Box, AppBar, Toolbar } from "@mui/material";

export function Navbar() {
    return (
        <nav>
            <Box>
                <AppBar>
                    <Toolbar sx={{display: 'flex', justifyContent: 'space-evenly'}}>
                        <NavLink to="/" element={<Home />} className="link">Home</NavLink>
                        <NavLink to="/signup" element={<Signup />}>Signup</NavLink>
                        <NavLink to="/admin" element={<Admin />} data-testid="adminLink">Admin</NavLink>
                    </Toolbar>
                </AppBar>
            </Box>

        </nav>
    )
}