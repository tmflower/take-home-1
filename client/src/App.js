import './App.css';
import { Routes, Route } from "react-router-dom";
import { Navbar } from "./Navbar";
import { Home } from "./routes/Home";
import { Signup } from "./routes/Signup";
import { Admin } from "./routes/Admin";
import { useState, useEffect } from "react";
import axios from 'axios';

function App() {
  const [users, setUsers] = useState([]);
  const [userAdded, setUserAdded] = useState(false);

  useEffect(() => {
    async function getUsers() {
      const usersList = await axios.get("http://localhost:3001/users");
      setUsers(usersList.data);
    }
    getUsers();
  }, [userAdded]);

  return (
    <div>
    <Navbar></Navbar>
    <Routes>
      <Route path="/" element={<Home />}></Route>
      <Route path="/signup" element={<Signup setUserAdded={setUserAdded}/>}></Route>
      <Route path="/admin" element={<Admin users={users}/>}></Route>
    </Routes>
    </div>

  );
}

export default App;
