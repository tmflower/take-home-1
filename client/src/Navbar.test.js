import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Navbar } from './Navbar';
import userEvent from "@testing-library/user-event";
import { App } from './App';

test('renders Home without crashing', () => {
    render(
        <MemoryRouter>
            <Navbar/>
        </MemoryRouter>);
});

test('matches snapshot', () => {
    const { asFragment } =
    render(
        <MemoryRouter>
            <Navbar/>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

test('signup link works', async() => {
    render(
        <MemoryRouter>
            <App />
        </MemoryRouter>
    );

    const headingBefore = screen.queryByText('Yodlr Registration Portal');
    expect(headingBefore).not.toBeInTheDocument();

    userEvent.click(screen.getByRole('link', { name: 'Signup' }));
    
    await waitFor(() => {
        const headingAfter = screen.getByText('Yodlr Registration Portal');
        expect(headingAfter).toBeInTheDocument();
    });
});

test('admin link works', async() => {
    render(
        <MemoryRouter>
            <App />
        </MemoryRouter>
    );

    const headingBefore = screen.queryByText('Admin Portal');
    expect(headingBefore).not.toBeInTheDocument();

    userEvent.click(screen.getByTestId('adminLink'));
    
    await waitFor(() => {
        const headingAfter = screen.getByText('Admin Portal');
        expect(headingAfter).toBeInTheDocument();
    });
});

test('home link works', async() => {
    render(
        <MemoryRouter>
                <App />
        </MemoryRouter>
    );

    userEvent.click(screen.getByRole('link', {name: 'Home'}));
    
    await waitFor(() => {
        const headingAfter = screen.getByText('Welcome to Yodlr!');
        expect(headingAfter).toBeInTheDocument();
    });
});