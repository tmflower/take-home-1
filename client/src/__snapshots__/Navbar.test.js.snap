// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1x5wx18-MuiToolbar-root"
        >
          <a
            aria-current="page"
            class="link active"
            element="[object Object]"
            href="/"
          >
            Home
          </a>
          <a
            class=""
            element="[object Object]"
            href="/signup"
          >
            Signup
          </a>
          <a
            class=""
            data-testid="adminLink"
            element="[object Object]"
            href="/admin"
          >
            Admin
          </a>
        </div>
      </header>
    </div>
  </nav>
</DocumentFragment>
`;
